corehq/form_processor/models.py

```diff
@@ -553,10 +557,19 @@ def get_data(self, path):
 
     @memoized
     def get_xml(self):
-        return self.get_attachment('form.xml')
+        try:
+            return self.get_attachment('form.xml')
+        except NotFound:
+            notify_exception(None, "Missing form XML", {
+                'form_id': self.form_id
+            })
+            return None
```

Comment:
I do not think this is a good change. Returning `None` is essentially an error code that needs to be evaluated at each level of the call stack. If the caller forgets to check a much more ambiguous `AttributeError: 'NoneType' object has no attribute ...` error will be raised somewhere later, which is much harder to catch and makes it more difficult to determine the real problem in Sentry if it is uncaught.

Raising a meaningful exception allows it to bubble up and be caught at the abstraction level where it makes sense to handle the error. It would be best (as similarly noted in [this comment](https://github.com/dimagi/commcare-hq/blob/a7e9c0fedacad6ae05ededc509deb5e5b5400567/corehq/form_processor/parsers/ledgers/form.py#L132)) to catch `NotFound` here and raise `MissingFormXML(form_id)`. I think in most cases missing form XML is a thing we want to know about in Sentry if the error is unexpected and uncaught.